### ISSUE TITLE
No Data: OPNSense

### ISSUE THREAD

**Issue by SticksDev** [pfelk/pfelk#495]
Hello! I've set everything up according to the docker guide, however I'm still not receiving data. I found a similar issue #490 - and added the roles and users because the docker guide doesn't mention them, but data still doesn't flow. Here's a screenshot of the log in OPNSense: 

![](https://img.sticks.ovh/Ai5v8UF1X)

Everything imported in Elastic..

![](https://img.sticks.ovh/lXjnIAici)

But still no indices...

![](https://img.sticks.ovh/v0udJCZek)

Ideas? Paging @a3ilson - you seem to be the only active responder now :/

**Comment by SticksDev** [comment]
Seems I found the root cause. Logstash is getting the logs, but it seems to failing to commicate with Search...

```
[WARN ] 2023-08-07 15:22:28.995 [Ruby-0-Thread-19: /usr/share/logstash/vendor/bundle/jruby/2.6.0/gems/logstash-output-elasticsearch-11.15.9-java/lib/logstash/outputs/elasticsearch/http_client/pool.rb:213] elasticsearch - Attempted to resurrect connection to dead ES instance, but got an error {:url=>"https://elastic:xxxxxx@es01:9200/", :exception=>LogStash::Outputs::ElasticSearch::HttpClient::Pool::BadResponseCodeError, :message=>"Got response code '401' contacting Elasticsearch at URL 'https://es01:9200/'"}```

**Comment by SticksDev** [comment]
I found it. Seems like there was some string issues when setting the passwords so ' was not omitted causing it to be invalid.

**Comment by SticksDev** [comment]
Weird. After rebooting the server with -d to detach from the output, it no longer sends data but I don't see any errors. 

Ideas @a3ilson?

**Comment by a3ilson** [comment]
My current setup is running via docker. I’ll need to test the pfelk/docker repo. The first thing that comes to mind when rebooting is ensuring the ‘sudo sysctl -w vm.max_map_count=262144’ is persistent. When checking docker it’ll appear to be working but in reality they are just resting in an endless loop.

**Comment by SticksDev** [comment]
Okay, I'm running this in a proxmox qmenu, but that should not matter. I've set the flag and am attempting to reboot the cluster now, standby.

**Comment by SticksDev** [comment]
Interesting, now I'm getting data from unbound, but not firewall:

![](https://img.sticks.ovh/26UL7JAdG)

**Comment by SticksDev** [comment]
But still no indices, :(

**Comment by a3ilson** [comment]
@SticksDev 
Update the pfelk.grok file and give it another try. I fixed a grok pattern issue where the pfSense would parse the OPNsense logs but if RFC5424 was utilized it failed to clean-up the dumb meta sequenceId values.

**Comment by a3ilson** [comment]
@SticksDev 
This should be fixed with the latest updates (23.08), reopen if the issue still exists. I would also consider ticking the RFC5424 within OPNsense.
